packages/x-data-grid/src/components/GridRow.tsx

```diff
@@ -533,7 +522,7 @@ const GridRow = React.forwardRef<HTMLDivElement, GridRowProps>(function GridRow(
       />
       {cells}
       {emptyCellWidth > 0 && <EmptyCell width={emptyCellWidth} />}
-      {rightCells.length > 0 && <div role="presentation" style={{ flex: '1' }} />}
+      {rightCells.length > 0 && <div role="presentation" className={gridClasses.filler} />}
```

Comment:
Moved inline styles to GridRootStyles since the filler is rendered for every row